vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -36,37 +37,55 @@ public abstract class LinkManager<L extends ILink<T>, T extends IVSpaceElement,
 
     @Autowired
     private IStorageEngine storage;
+    
+    @Autowired
+    private IImageService imageService;
+
 
     @Override
     public U createLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
-            String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename)
-            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+            String linkLabel, String linkDesc, DisplayType displayType, byte[] linkImage, String imageFilename, String existingImageId)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException, ImageDoesNotExistException {
 
         L link = createLinkObject(title, id);
         T target = getTarget(linkedId);
         link.setName(linkLabel);
+        link.setDescription(linkDesc);
         link.setTarget(target);
         U displayLink = createDisplayLink(link);
-        setDisplayProperties(displayLink, id, positionX, positionY, rotation, displayType, linkImage, imageFilename);
+        if(existingImageId!=null && !existingImageId.trim().isEmpty()) {
+            setDisplayProperties(displayLink, positionX, positionY, rotation, displayType, existingImageId);
+        }
+        else {
+            setDisplayProperties(displayLink, id, positionX, positionY, rotation, displayType, linkImage, imageFilename);
+        }
         return updateLinkAndDisplay(link, displayLink);
 
     }
 
     @Override
     public U updateLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
-            String linkLabel, String linkId, String linkDisplayId, DisplayType displayType, byte[] linkImage,
-            String imageFilename)
-            throws SpaceDoesNotExistException, LinkDoesNotExistsException, ImageCouldNotBeStoredException {
+            String linkLabel, String linkDesc,String linkId,  String linkDisplayId, DisplayType displayType, byte[] linkImage,
```

Comment:
there should be one space between `,String` and only one space after the comma (`,  String`)